File: edu/TUD/TI1406 Computerorganisatie/pseudo/power.java
/*
* The pow subroutine calculates powers of natural bases
* and exponents.
*
* Arguments:
*
* base - the exponential base
* exp - the exponent
*
* Return value: 'base' raised to the power of 'exp'.
*/
class power
{
	public static void main(String[] args)
	{
		System.out.println(pow(2, 10)); // = 1024
	}
	
	public static int pow(int base, int exponent)
	{
		int sum = 1;
		
		while (exponent > 0)
		{
			sum *= base;
			exponent--;
		}

		return sum;
	}
}

File: edu/TUD/TI1406 Computerorganisatie/pseudo/power.c
#include <stdio.h>

long int power(int, int);

int main()
{
	puts("Assignment 4: power");
	printf("%ld\n", power(2, 10));
	return 0;
}

long int power(int base, int exp)
{
	int total = 1;
	
	while (exp > 0)
	{
		exp--;
		total *= base;
	}
	
	return total;
}

